### ISSUE TITLE
Error in the softmax layer of Matconvnet

### ISSUE THREAD

**Issue by jebaberlin** [vlfeat/matconvnet#590]
Iam trying to train the network of my own using cnn_train comment in the Matconvnet .The input dataset consists of four classes and 15 training images for each class which is sored as given below
imagedatabase = 

    images: [1x1 struct]
      meta: [1x1 struct]
imagedatabase.images

ans = 

     data: [64x64x60 single]
    label: [1x60 double]
      set: [1x60 double]
imagedatabase.meta

ans = 

    classes: '1234'
       sets: {'train'  'val'}

The network is initialized as given below
f=1/100 ;
net.layers = {} ;
net.layers{end+1} = struct('type', 'conv', ...
                           'filters', f*randn(3,3,1,64, 'single'), ...
                           'biases', zeros(1, 64, 'single'), ...
                           'stride', 1, ...
                           'pad', 1) ;
net.layers{end+1} = struct('type', 'relu') ;
net.layers{end+1} = struct('type', 'pool', ...
                           'method', 'max', ...
                           'pool', [2 2], ...
                           'stride', 2, ...
                           'pad', 0) ;
net.layers{end+1} = struct('type', 'conv', ...
                           'filters', f*randn(1,1,64,20, 'single'), ...
                           'biases', zeros(1,20, 'single'), ...
                           'stride', 1, ...
                           'pad', 1) ;
net.layers{end+1} = struct('type', 'relu') ;
net.layers{end+1} = struct('type', 'conv', ...
                           'filters', f*randn(1,1,20,4, 'single'), ...
                           'biases', zeros(1,4, 'single'), ...
                           'stride', 1, ...
                           'pad', 0) ;
 net.layers{end+1} = struct('type', 'softmax') ;  

trainOpts.batchSize = 1 ;
trainOpts.numEpochs = 1 ;
trainOpts.continue = true ;
trainOpts.useGpu = false ;
trainOpts.learningRate = 0.001 ;

The program flow goes as 
imageMean = mean(imagedatabase.images.data(:)) ;
imagedatabase.images.data = imagedatabase.images.data - imageMean ;
% Call training function in MatConvNet
[net,info] = cnn_train(net, imagedatabase, @getBatch, trainOpts) ;

When the program is executed the error is obtained as

learning rate changed (0.000000 --> 0.001000): resetting momentum
training: epoch 01: processing batch   2 of  40 ...Error using  .* 
Array dimensions must match for binary array op.

Error in vl_nnsoftmax (line 34)
Y = Y .* bsxfun(@minus, dzdY, sum(dzdY .* Y, 3)) ;

Error in vl_simplenn (line 211)
        res(i).dzdx = vl_nnsoftmax(res(i).x, res(i+1).dzdx) ;

Error in cnn_train (line 140)
    res = vl_simplenn(net, im, one, res, ...

Error in mnl_eg (line 34)
[net,info] = cnn_train(net, imagedatabase, @getBatch, trainOpts) ;

In some issues they have mentioned to use softmaxloss instead of softmax while training but still I receive the error as
learning rate changed (0.000000 --> 0.001000): resetting momentum
training: epoch 01: processing batch   2 of  40 ...Error using vl_nnsoftmaxloss (line 42)
Assertion failed.

Error in vl_simplenn (line 163)
      res(i+1).x = vl_nnsoftmaxloss(res(i).x, l.class) ;

Error in cnn_train (line 140)
    res = vl_simplenn(net, im, one, res, ...

Error in mnl_eg (line 34)
[net,info] = cnn_train(net, imagedatabase, @getBatch, trainOpts) ;

Iam very thankful if the above error is rectified. 
I have gone through "error on cnn_minst #51"  but I couldn't get to clear the error
Thanks in advance
